src/ui/shared/UrlContext.ts

```diff
@@ -0,0 +1,82 @@
+export type AppMode = 'onboarding' | 'wallet';
+export type WindowType = 'popup' | 'tab' | 'dialog';
+export type WindowLayout = 'column' | 'page';
+
+export enum UrlContextParam {
+  windowType = 'windowType',
+  windowLayout = 'windowLayout',
+  appMode = 'appMode',
+}
+
+function getSearchParam<T>(param: UrlContextParam, defaultValue: T) {
+  const url = new URL(window.location.href);
+  return (url.searchParams.get(param) as T) || defaultValue;
+}
+
+export interface UrlContextParams {
+  appMode?: AppMode;
+  windowType?: WindowType;
+  windowLayout?: WindowLayout;
+}
+
+class UrlContext {
```

Comment:
Why are these classes? They do not hold state

Reply:
Ah, they did, just one commit before... Post-refactoring cringe moments, as always. I should come up with smth better for this

Reply:
I simplified the `UrlContext.ts`, please check out the new version